sumologic/resource_sumologic_cse_match_list.go

```diff
@@ -0,0 +1,307 @@
+			DefaultTtl:   d.Get("default_ttl").(int),
+			Description:  d.Get("description").(string),
+			Name:         d.Get("name").(string),
+			TargetColumn: d.Get("target_column").(string),
+		})
+
+		if err != nil {
+			return err
+		}
+		d.SetId(id)
+
+		//Match list items
+		itemsData := d.Get("items").([]interface{})
+		var items []CSEMatchListItemPost
+		for _, data := range itemsData {
+			item, _ := resourceToCSEMatchListItem([]interface{}{data})
+			items = append(items, item)
+
+		}
+
+		if len(items) > 0 {
+			err2 := c.CreateCSEMatchListItems(items, id)
+			if err2 != nil {
+				log.Printf("[WARN] An error occurred while adding match list items to match list id: %s, err: %v", id, err2)
+			}
+
+			// Calling Sleep method, adding items might take a while before items retrieved in next section
+			// Need to find a better way since feels super hacky
+			time.Sleep(15 * time.Second)
+
+		}
+
+	}
+
+	return resourceSumologicCSEMatchListRead(d, meta)
+}
+
+func resourceToCSEMatchListItem(data interface{}) (CSEMatchListItemPost, string) {
+	itemsSlice := data.([]interface{})
+	item := CSEMatchListItemPost{}
+	if len(itemsSlice) > 0 {
+		itemObj := itemsSlice[0].(map[string]interface{})
+		item.ID = itemObj["id"].(string)
+		item.Description = itemObj["description"].(string)
+		item.Active = true
+		item.Expiration = itemObj["expiration"].(string)
+		item.Value = itemObj["value"].(string)
+	}
+	return item, item.ID
+}
+
+func resourceSumologicCSEMatchListUpdate(d *schema.ResourceData, meta interface{}) error {
+	CSEMatchListPost, err := resourceToCSEMatchList(d)
+	if err != nil {
+		return err
+	}
+
+	c := meta.(*Client)
+	if err = c.UpdateCSEMatchList(CSEMatchListPost); err != nil {
+		return err
+	}
+
+	//Match list items
+	itemsData := d.Get("items").([]interface{})
+	var itemIds []string
+	var items []CSEMatchListItemPost
+	for _, data := range itemsData {
+		item, id := resourceToCSEMatchListItem([]interface{}{data})
+		items = append(items, item)
+		itemIds = append(itemIds, id)
+
+	}
+
+	if len(items) > 0 {
+		for _, item := range items {
+			CSEMatchListItem, er := c.GetCSEMatchListItem(item.ID)
+			log.Printf("[WARN] An error occurred while getting match list item with id: %s, err: %v", item.ID, er)
+			if CSEMatchListItem != nil {
+				if contains(itemIds, CSEMatchListItem.ID) {
+					err3 := c.UpdateCSEMatchListItem(item)
+					if err3 != nil {
+						log.Printf("[WARN] An error occurred while updating match list item with id: %s, err: %v", item.ID, err3)
+					}
+				} else {
+					err3 := c.DeleteCSEMatchListItem(CSEMatchListItem.ID)
+					if err3 != nil {
+						log.Printf("[WARN] An error occurred deleting match list item with id: %s, err: %v", CSEMatchListItem.ID, err3)
+					}
+				}
+			} else {
+				err4 := c.CreateCSEMatchListItems(items, d.Id())
+				if err4 != nil {
+					log.Printf("[WARN] An error occurred while adding match list items to match list id: %s, err: %v", d.Id(), err4)
+				}
+			}
+		}
+	} else {
```

Comment:
I think we don't need the else. This block here which deletes items that aren't supposed to be in the list anymore needs to run regardless of how many list items there are.

Reply:
I see your point, done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/353/commits/1fd51d0bb44378ea7406d45f332bfe8faa5e877b